tests/e2e/config_secret_utils.go

```diff
@@ -0,0 +1,331 @@
+		"govc role.remove CNS-HOST-CONFIG-STORAGE-" + testUser + ";" +
+		"govc role.remove CNS-VM-" + testUser + ";" +
+		"govc role.remove CNS-SEARCH-AND-SPBM-" + testUser
+	framework.Logf(deleteRoles)
+	result, err = sshExec(sshClientConfig, masterIp, deleteRoles)
+	if err != nil && result.Code != 0 {
+		fssh.LogResult(result)
+		return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			deleteRoles, masterIp, err)
+	}
+
+	// delete test user
+	deleteUser := loginCmd + "govc sso.user.rm " + testUser
+	framework.Logf(deleteUser)
+	result, err = sshExec(sshClientConfig, masterIp, deleteUser)
+	if err != nil && result.Code != 0 {
+		fssh.LogResult(result)
+		return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			deleteUser, masterIp, err)
+	}
+	return nil
+}
+
+func createRolesForTestUser(loginCmd string, sshClientConfig *ssh.ClientConfig,
+	masterIp string, testUser string) error {
+	createRoleCmdFortestUser := loginCmd + "govc role.create CNS-DATASTORE-" + testUser +
+		" Datastore.FileManagement;" +
+		"govc role.create CNS-HOST-CONFIG-STORAGE-" + testUser +
+		" Host.Config.Storage;" +
+		"govc role.create CNS-VM-" + testUser +
+		" VirtualMachine.Config.AddExistingDisk VirtualMachine.Config.AddRemoveDevice;" +
+		"govc role.create CNS-SEARCH-AND-SPBM-" + testUser +
+		" Cns.Searchable StorageProfile.View"
+	framework.Logf(createRoleCmdFortestUser)
+	result, err := sshExec(sshClientConfig, masterIp, createRoleCmdFortestUser)
+	if err != nil && result.Code != 0 {
+		fssh.LogResult(result)
+		return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			createRoleCmdFortestUser, masterIp, err)
+	}
+	return nil
+}
+
+func getDataCenterClusterHostAndVmDetails(loginCmd string, sshClientConfig *ssh.ClientConfig,
+	masterIp string) (string, string, error) {
+	// fetch datacenter details
+	ginkgo.By("Executing command for fetching data center details")
+	dataCenter := loginCmd + "govc datacenter.info | grep -i Name | awk '{print $2}' | tr -d '\n'"
+	framework.Logf(dataCenter)
+	dcResult, err := sshExec(sshClientConfig, masterIp, dataCenter)
+	if err != nil && dcResult.Code != 0 {
+		fssh.LogResult(dcResult)
+		return "", "", fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			dataCenter, masterIp, err)
+	}
+
+	// fetch cluster details
+	ginkgo.By("Executing command for fetching cluster details")
+	cluster := loginCmd + "govc ls /" + dcResult.Stdout + "/host | tr -d '\n'"
+	framework.Logf(cluster)
+	clusterResult, err := sshExec(sshClientConfig, masterIp, cluster)
+	if err != nil && clusterResult.Code != 0 {
+		fssh.LogResult(clusterResult)
+		return "", "", fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			cluster, masterIp, err)
+	}
+
+	// Get all esxi hosts details
+	ginkgo.By("Executing command for fetching esxi host details")
+	hosts := loginCmd + "govc ls " + clusterResult.Stdout + " " + " | grep 10. > /root/hosts.txt | tr -d '\n'"
```

Comment:
don't create a file, instead call the smaller subroutine each time this info is required or fetch this info once for a testrun and store it in some variable instead

Reply:
done